server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1785,42 +1964,62 @@ void handleSelfCertificate(Consumer consumer, Pool pool,
      */
     private class NewHandler implements EntitlementHandler{
         @Override
-        public Entitlement handleEntitlement(Consumer consumer, Pool pool,
-            Entitlement entitlement, int quantity) {
-            Entitlement newEntitlement = new Entitlement(pool, consumer, quantity);
-            consumer.addEntitlement(newEntitlement);
-            pool.getEntitlements().add(newEntitlement);
-            return newEntitlement;
+        public Map<String, Entitlement> handleEntitlement(Consumer consumer,
+                Map<String, PoolQuantity> poolQuantities,
+                Map<String, Entitlement> entitlements) {
+            Map<String, Entitlement> result = new HashMap<String, Entitlement>();
+            for (Entry<String, PoolQuantity> entry : poolQuantities.entrySet()) {
+                Entitlement newEntitlement = new Entitlement(entry.getValue().getPool(), consumer, entry
+                        .getValue().getQuantity());
+                result.put(entry.getKey(), newEntitlement);
+                consumer.addEntitlement(newEntitlement);
```

Comment:
@nguyenfilip This worked brilliantly. thanks for the help! fixed in the merged PR.